sounddevice.py

```diff
@@ -2266,9 +2266,27 @@ def reset(self):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
-
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors. 
+
+    Attributes
+    ----------
+    paerrorcode : int or None
+        The PaErrorCode integer value or None if there is no associated error
+        code. Note that some PortAudioError's do not have an associated
+        PaErrorCode. In such a case `paerrorcode` will be None.
+    """
+    def __init__(self, *args):
+        if len(args) > 2:
+            raise TypeError("PortAudioError expected at most 2 arguments, got %d" % len(args))
+        elif len(args) == 2:
+            self.paerrorcode, errortext = args
+            errortext = "[PaErrorCode {0}] {1}".format(self.paerrorcode, errortext)
```

Comment:
Here the low-level error number comes before the high-level error text (for which it is actually meaningless).
I think I would prefer the `[PaErrorCode XYZ]` coming at the very end, after the low-level error message. Also, casual users will not be very interested in the error code, they should see the friendlier error message first.